horace_core/algorithms/private/gen_sqw_check_files.m

```diff
@@ -99,43 +99,122 @@
 end
 
 
-% Check parameter file
+% Check parameter files
 % --------------------
-if ~isempty(par_file)
-    if is_string(par_file) && ~isempty(strtrim(par_file))
-        par_file_out=strtrim(par_file);
-        det_par_file = true;
-    elseif isstruct(par_file)
-        det_par_file = false;
-    else
-        error('HORACE:algorithms:invalid_argument',...
-            'If given, par filename  must be a non-empty string');
-    end
-    if det_par_file
-        % Check par file exists
-        [~,~,ext]=fileparts(par_file_out);
-        if any(strcmpi(ext,[ext_horace,'.spe']))
-            error('HORACE:algorithms:invalid_argument',...
-                ['Detector parameter files must not have the reserved extension: "%s". ',...
-                '''. Check the file is .par type and rename.'],ext);
+
+% Ensure par_file is a cell array with the same number of elements as the
+% number of spe files. A single input .par file name is expanded to as many
+% duplicates as required; an empty file name is expanded to duplicates of
+% the empty string
+n_par_files = numel(spe_file_out); % get nr par files from nr spe files
+if isempty(par_file)
+    % det_par_file indicates if the relevant element represents a filename
+    % here it is false as the contents are empty
+    % the single value is converted to a cell of empty chars, one per spe
+    det_par_file = false(n_par_files,1);
+    par_file_out = repmat({''},n_par_files,1);
+elseif istext(par_file) 
+    % a single text string has been provided
+    % det_par_file indicates that all the pare_file_outs will be filenames
+    % the single string is converted to a cell of repeated values
+    det_par_file = true(n_par_files,1);
+    par_file_out = repmat({strtrim(par_file)},n_par_files,1);
+elseif isstruct(par_file) && numel(par_file) == 1
+    % a single struct has been provided
+    % det_par_file is false for all spes as a filename has not been
+    % provided. The struct is repeated one for each spe
+    det_par_file = false(n_par_files,1);
+    % the efficiency of this presumes that copy on write will mean that
+    % only one copy of the struct will actually be in memory
+    par_file_out = repmat({ par_file },n_par_files,1);
+elseif iscell(par_file) 
+    if numel(par_file) == 1 
```

Comment:
```suggestion
    if isscalar(par_file)
```

Reply:
true but not quite saying, just one par_file. The present version gives better intent